plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -391,9 +638,101 @@ func (c *CiscoTelemetryMDT) parseKeyField(tags map[string]string, field *telemet
 	}
 }
 
+func (c *CiscoTelemetryMDT) parseRib(grouper *metric.SeriesGrouper, field *telemetry.TelemetryField, prefix string, path string, tags map[string]string, timestamp time.Time) {
+        // RIB 
+        measurement := path
+        for _, subfield := range field.Fields {
+            //For Every table fill the keys which are vrfName, address and masklen
+            if subfield.Name == "vrfName" {
+                tags["vrfName"] = decodeTag(subfield)
+            }
+            if subfield.Name == "address" {
+                tags["address"] = decodeTag(subfield)
+            }
+            if subfield.Name == "maskLen" {
+                tags["maskLen"] = decodeTag(subfield)
+            }
+            if value := decodeValue(subfield); value != nil {
+                grouper.Add(measurement, tags, timestamp, subfield.Name, value)
+            }
```

Comment:
Can you really rely on the order of `subfield.Name`s? Otherwise, depending on the order in `field.Fields`, you might emit metrics with different number of tags.